### PR TITLE
Adds named parameters to exception method

### PR DESCRIPTION
**Pull request by JordiPolo**
I see people want to change the level of the exception log but 
`logger.exception(e, nil, nil, :warn)` looks really obscure

What do you think about this interface instead?

@jfeltesse-mdsol  @ssteeg-mdsol  @ykitamura-mdsol

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -50,10 +50,10 @@ task :benchmark do
   simple_log = create_simple_logger
 
   Benchmark.ips do |bm|
-    bm.report('short content') { log.error(contents) }
-    bm.report('Logger short content') { simple_log.info(contents) }
-    bm.report('long content') { log.info(long_contents) }
-    bm.report('Logger long content') { simple_log.info(long_contents) }
+    bm.report('JSON short content') { log.debug(contents) }
```
Comment (JordiPolo) [inline Rakefile]: Unrelated. Clearer output this way. Only for dev so does not matter much.

**Comment by jfeltesse-mdsol** [comment]
changes look good to me. 

are you planning on supporting both styles long term? if not another way would be to make it named params only and cut 2.0 with that breaking change

File: spec/lib/lorekeeper/json_logger_spec.rb [hunk 2]
```diff
@@ -109,6 +115,25 @@
           end
         end
 
+        context 'Logging an exception with custom level' do
+          let(:exception_data) do
```
Comment (ssteeg-mdsol) [inline spec/lib/lorekeeper/json_logger_spec.rb]: Couldn't it just be one `merge`?

```rb
base_message.merge(
  'exception' => "StandardError: #{exception_msg}",
  'message' => exception_msg,
  'stack' => stack,
  'level' => 'info'
)

File: spec/lib/lorekeeper/json_logger_spec.rb [hunk 3]
```diff
@@ -271,6 +302,10 @@
             logger.remove_thread_unsafe_fields(['planet'])
           end
           it_behaves_like 'Logging methods'
+
+          it 'removes information from the local thread' do
+            expect(Thread.current[Lorekeeper::JSONLogger::THREAD_KEY]).to eq(nil)
```
Comment (ssteeg-mdsol) [inline spec/lib/lorekeeper/json_logger_spec.rb]: `.to eq(nil)` → `.to be_nil`

File: spec/lib/lorekeeper/json_logger_spec.rb [hunk 4]
```diff
@@ -162,15 +193,15 @@
...
+                'data' => { 'planet' => 'hyperion' },
+                'message' => "String: #{message.inspect} second part not good",
                 'timestamp' => time_string
               }
             ]
           end
 
           it 'Logs the exception message' do
-            logger.exception(message)
+            logger.exception(message, message: 'second part not good', data: {'planet' => 'hyperion' })
```
Comment (ssteeg-mdsol) [inline spec/lib/lorekeeper/json_logger_spec.rb]: space missing in the hash: `data: { 'planet'`

File: lib/lorekeeper/json_logger.rb [hunk 5]
```diff
@@ -60,26 +60,33 @@ def remove_fields(fields)
       end
     end
 
+
```
Comment (ssteeg-mdsol) [inline lib/lorekeeper/json_logger.rb]: extra line

**Comment by JordiPolo** [comment]
@jfeltesse-mdsol 2.0 is a good idea.
Let's release this, let it be used a couple of months, see what reception it has and then we can cut a version deleting the old interface.

**Pull request merged**